src/main/java/org/openstreetmap/atlas/tags/annotations/extraction/LayerExtractor.java

```diff
@@ -0,0 +1,57 @@
+package org.openstreetmap.atlas.tags.annotations.extraction;
+
+import java.util.Optional;
+
+import org.openstreetmap.atlas.tags.BridgeTag;
+import org.openstreetmap.atlas.tags.LayerTag;
+import org.openstreetmap.atlas.tags.Taggable;
+import org.openstreetmap.atlas.tags.TunnelTag;
+
+/**
+ * Extracts layer values.
+ *
+ * @author bbreithaupt
+ */
+public final class LayerExtractor
+{
+    /**
+     * Extracts layer values as {@link Optional} {@link Long}s based on the {@link LayerTag},
+     * {@link BridgeTag}, and {@link TunnelTag}s. Defaults to 0 if no layer value is found.
+     *
+     * @param taggable
+     *            {@link Taggable} to get a value for
+     * @return {@link Optional} {@link Long} layer value. Will return empty if taggable has a
+     *         {@link LayerTag} that is invalid.
+     */
+    public static Optional<Long> validateAndExtract(final Taggable taggable)
```

Comment:
Looks like we need to include the tunnel and bridge logic in the [areOnSameLayer()] method in LayerTag.(https://github.com/osmlab/atlas/blob/0b6cdf02255ca4f51048faaab72490544b19a196/src/main/java/org/openstreetmap/atlas/tags/LayerTag.java#L35). Currently it defaults to 0 if no layer tag is present.

Reply:
Now that we are using `LayerTag.getTaggedOrImpliedValue()` this is handled.